app/services/get_information_from_login_key.rb

```diff
@@ -0,0 +1,44 @@
+class GetInformationFromLoginKey
```

Comment:
We have been a bit inconsistent with how we name services, we can call them with nouns like `SubscriptionFinder`, or with a verb like `GetSubjectName`. You choose the latter, later on we could decide if we want nouns or verbs.